cpp/src/arrow/memory_pool.cc

```diff
@@ -87,6 +87,19 @@ const char* je_arrow_malloc_conf =
 #endif  // ARROW_JEMALLOC
 
 namespace arrow {
+
+std::vector<std::string> SupportedBackendNames() {
```

Comment:
It would be better to reuse `SupportedBackends` below. Otherwise we risk this function being out of sync.

Reply:
I had tried but was failing on some namespace issues; can try again.

Reply:
I fixed this and renamed it to `SupportedMemoryBackendNames`